modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/ModifyNode.java

```diff
@@ -133,7 +132,8 @@ public ModifyNode(
                         .mapToInt(ColumnDescriptor::logicalIndex)
                         .toArray();
 
-        this.rowFactory = rowFactory;
+        RowSchema rowSchema = rowSchemaForTableDescriptor(table.descriptor());
```

Comment:
I checked GG9 with spilling enabled in ItDmlTest and looks like that's not correct schema :thinking: 
At least for Merge.